source/material_model/damage_rheology.cc

```diff
@@ -0,0 +1,2027 @@
+              out.specific_heat[i] = specific_heat(in.temperature[i], pressure, in.composition[i], in.position[i]);
+            }
+
+          out.thermal_expansion_coefficients[i] = std::max(std::min(out.thermal_expansion_coefficients[i],max_thermal_expansivity),min_thermal_expansivity);
+          out.specific_heat[i] = std::max(std::min(out.specific_heat[i],max_specific_heat),min_specific_heat);
+        }
+    }
+
+
+
+    template <int dim>
+    void
+    DamageRheology<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Damage rheology model");
+        {
+          prm.declare_entry ("Reference density", "3300",
+                             Patterns::Double (0),
+                             "Reference density $\\rho_0$. Units: $kg/m^3$.");
+          prm.declare_entry ("Reference temperature", "293",
+                             Patterns::Double (0),
+                             "The reference temperature $T_0$. Units: $K$.");
+          prm.declare_entry ("Viscosity", "5e24",
+                             Patterns::Double (0),
+                             "The value of the constant viscosity. Units: $kg/m/s$.");
+          prm.declare_entry ("Composition viscosity prefactor 1", "1.0",
+                             Patterns::Double (0),
+                             "A linear dependency of viscosity on the first compositional field. "
+                             "Dimensionless prefactor. With a value of 1.0 (the default) the "
+                             "viscosity does not depend on the composition.");
+          prm.declare_entry ("Composition viscosity prefactor 2", "1.0",
+                             Patterns::Double (0),
+                             "A linear dependency of viscosity on the second compositional field. "
+                             "Dimensionless prefactor. With a value of 1.0 (the default) the "
+                             "viscosity does not depend on the composition.");
+          prm.declare_entry ("Compositional density difference", "100.0",
+                             Patterns::Double (),
+                             "Density excess of the first compositional field."
+                             "Units: $kg/m^3$");
+          prm.declare_entry ("Thermal conductivity", "4.7",
+                             Patterns::Double (0),
+                             "The value of the thermal conductivity $k$. "
+                             "Units: $W/m/K$.");
+          prm.declare_entry ("Reference specific heat", "1250",
+                             Patterns::Double (0),
+                             "The value of the specific heat $cp$. "
+                             "Units: $J/kg/K$.");
+          prm.declare_entry ("Thermal expansion coefficient", "2e-5",
+                             Patterns::Double (0),
+                             "The value
```

Comment:
Is that even used anywhere? I couldn't find it...

Reply:
Yes, it is stored in the variable `thermal_alpha` and used if `use_table_properties` is `false`.

Reply:
Somehow github messed this up. The comment is about line 1551.

Reply:
prm.declare_entry ("Corresponding phase for transition", "",